common/remote_util.cc

```diff
@@ -47,10 +49,29 @@ void RemoteUtil::SetScpCommand(std::string scp_command) {
   scp_command_ = std::move(scp_command);
 }
 
+void RemoteUtil::SetSftpCommand(std::string sftp_command) {
+  sftp_command_ = std::move(sftp_command);
+}
+
 void RemoteUtil::SetSshCommand(std::string ssh_command) {
   ssh_command_ = std::move(ssh_command);
 }
 
+// static
+std::string RemoteUtil::ScpToSftpCommand(std::string scp_command) {
+  // "scp", "winscp.exe", "C:\path\to\scp", "/scppath/scp --foo" etc.
+  size_t pos = 0;
+  while ((pos = scp_command.find("scp", pos)) != std::string::npos) {
+    const char next_ch = scp_command[pos + 3];
+    if ((next_ch == 0 || next_ch == '.' || next_ch == ' ')) {
+      return scp_command.replace(pos, 3, "sftp");
```

Comment:
Suggestion:
- If the path contains `scp`, replace it with `sftp` and check if it exists.
- If it does not exist, just try replacing the entire (absolute or relative) binary path with `sftp` and hope it can be found in the path.

Reply:
It's not trivial to determine which part the path is and which part is the rest. You might have C:\Some -Path\to the\Open SSH\scp.exe. CreateProcess actually handles that (without quotes!), see https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessa. So to get this right, I'd need a pretty sophisticated logic. I didn't want to go down that path, but wanted to handle at least some stupid (but not super unlikely) corner cases, e.g. if "scp" is in someone's user name like C:\Users\chriscpine\OpenSSH\scp.exe.

Maybe I'll just print a warning if the converted scp is used and let users know how to fix it. WDYT?

Reply:
What I don't like about this transparent conversion to sftp is that if the binary cannot be found at the resulting path, CreateProcess fails and the user receives an error message for a binary path that she never entered and might be scratching her head where that path comes from. So a warning should be issued every time the deprecated flags/vars are detected.

Reply:
Yes, this will be the case. My previous comment contains the warning that will be printed if the user runs cdc_rsync or cdc_stream with the scp flags. I decided to put the warnings into higher level code, though (that actually know about the flags). It didn't feel right to print it here. Therefore, the warning is not included in this CL.